components/survey/List/index.jsx

```diff
@@ -0,0 +1,90 @@
+import Slider from 'react-slick';
+import { useEffect, useState } from 'react';
+import { PropTypes } from 'prop-types';
+
+import SurveyCard from '@/components/survey/Card';
+
+import useUser from 'hooks/useUser';
+import useSurveys from 'hooks/useSurveys';
+import { sliderSettings } from './sliderSettings';
+import { formatDate } from 'helpers/date';
+
+import 'slick-carousel/slick/slick.css';
+import 'slick-carousel/slick/slick-theme.css';
+import styles from './index.module.scss';
+
+export const surveyListTestIds = {
+  base: 'survey-list',
+  date: 'survey-list__date',
+  text: 'survey-list__text',
+};
+
+const SurveyList = ({ setBackgroundImagePath }) => {
+  const [surveys, setSurveys] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [slideId, setSlideId] = useState();
+
+  const date = new Date();
+  const { user } = useUser();
+  const surveyResponse = useSurveys(user, currentPage);
```

Comment:
```suggestion
  const surveys = useSurveys(user, currentPage);
```

Should we move more logic into the `useSurveys` hook? 🤔  
I am thinking about:
- Replace the `totalPages, setTotalPages` logic by a `surveys.length` — so no need to manage it manually
- moving the `updatedSurveyResponse` mechanism (adding the newly parsed survey) into the hook

The main idea is to have more explicit responsibilities: 
- The `useSurveys` hook provides you with surveys and handles the changes in the surveys. 
- The `SurveyList` component is in charge to display a list of these surveys — but not to manipulate them

> In practice, that means the SurveyList component should not edit the content of `surveys` but only be re-rendered when this list changes (from the hook).

Let me know if there is a blocker that I did not identify 🙌

Reply:
I really like the idea of separating data fetching and the representation of data in `hooks` and `components` respectively!
With that in mind, I made slight changes in the implementation:

Initially, the user would see the number of dots equal to the total records. Then on clicking a dot, if data isn't fetched yet for that dot, we calculate the page number from the dot index and fetch data for that particular page.

This way users won't be surprised to see the number of dots getting updated after each slide changes!

`useSurveys` implementation isn't changed. The reason is:
- Fetching all data in `useSurveys` IMHO might take a lot of time. The current implementation only fetches data for a single page and returns!
- To make use of the `useSwr` caching,  the `currentPage` parameter acts as a cache key in the `useSwr` hook!

Thank you!

Reply:
Looks good to me! :) I'm still wondering if there is any easier way, but I cannot think of anything else ^^, So all good :)